src/generator.rs

```diff
@@ -96,9 +97,50 @@ pub fn run_generator(devices: &[Device], output_directory: &Path, fake_mode: boo
         }
     }
 
+    let compressors: BTreeSet<&String> = devices
+        .iter()
+        .filter(|device| device.compression_algorithm.is_some())
+        .flat_map(|device| device.compression_algorithm.as_ref())
+        .collect::<BTreeSet<_>>();
+
+    if !compressors.is_empty() {
+        let known = get_known_compressors()?;
+
+        for comp in compressors.into_iter().filter(|&c| !known.contains(c)) {
```

Comment:
This is `compressors.difference(&known)`. Granted, that will not work out of box since one is `BTreeSet<&String>` and the other is `BTreeSet<String>`.

If I were to go full shitlord on this, I'd lift the `fs::read_to_string()` to be parallel to `known`, pass it to `get_known_compressors()` (now known as `parse_known_compressors()` or sth), and make the compressors flatmap be over `d.c_a.as_ref().map(String::as_str)` (or `d.c_a.as_ref().map(|a| &a[..])` as the case may be) to have them both be `BTreeSet<&str>` (and save on some needless allocations) to spell this out correctly semantically.

Especially since the `get_known_compressors()`/`_get_known_compressors()` looks like it does now, as opposed to how it did last time around.